impl/src/main/java/impl/QuestionResourceImpl.java

```diff
@@ -89,17 +92,40 @@ public QuestionResourceImpl(QuestionDao questionDao, QuestionVoteDao questionVot
 
     @Override
     public Observable<List<Question>> getLatestQuestions(Integer limit) {
-        limit = firstNonNull(limit, 10);
+        return this.questionDao.getLatestQuestions(firstNonNull(limit, 10)).toList()
```

Comment:
I actually preferred it when firstNonNull(limit, 10) was on it's own row, because it separates logic of finding the limit from fetching the questions. Shorter simpler rows, in short :)
Change as you please, just voicing my opinions :D

Reply:
We might actually want to use something else than firstNonNull to prevent too large values too, but that is out of scope here. Just saying for next time we're touching this stuff.

Reply:
I guess both default and max could be set with annotations.